libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/eh/RoadObjectMatcher.kt

```diff
@@ -0,0 +1,138 @@
+package com.mapbox.navigation.core.trip.session.eh
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.factory.EHorizonInstanceFactory
+import com.mapbox.navigation.base.internal.factory.RoadObjectInstanceFactory
+import com.mapbox.navigation.base.trip.model.eh.OpenLRStandard
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigator
+import com.mapbox.navigator.RoadObject
+import com.mapbox.navigator.RoadObjectMatcherError
+import com.mapbox.navigator.RoadObjectMatcherListener
+
+private typealias SDKRoadObject =
+    com.mapbox.navigation.base.trip.model.roadobject.RoadObject
+
+private typealias SDKRoadObjectMatcherError =
+    com.mapbox.navigation.base.trip.model.roadobject.RoadObjectMatcherError
+
+/**
+ * [MapboxNavigation.roadObjectMatcher] provides methods to match custom road objects.
+ */
+class RoadObjectMatcher internal constructor(
+    private val navigator: MapboxNativeNavigator
+) {
+
+    /**
+     * Road object matching listener.
+     */
+    var roadObjectMatcherObserver: RoadObjectMatcherObserver? = null
+        set(value) {
+            if (value != null) {
+                navigator.roadObjectMatcher?.setListener(roadObjectMatcherListener)
```

Comment:
Let's document that the `roadObjectMatcherObserver` needs to be registered before any of the other methods are called. Otherwise, the results are lost.

Reply:
done